provision/README.md

```diff
@@ -1,43 +1,149 @@
-# PROVISION service
+# Provision service
 
-PROVISION service provides an HTTP API to interact with Mainflux.
+Provision service provides an HTTP API to interact with [Mainflux][mainflux]. 
```

Comment:
General observation: break long chunks of text in paragraphs, separate paragraphs with whitelines - it will be much easier to read.